src/osm_configurator/model/project/calculation/attractivity_phase.py

```diff
@@ -25,4 +55,108 @@ def calculate(self, configuration_manager):
         Returns:
             calculation_state_enum.CalculationState: The state of the calculation, after this phase finished its execution or failed trying so.
         """
-        pass
+        # Get the traffic cells
+        geojson_path: Path = configuration_manager.get_cut_out_configuration().get_cut_out_path()
+        co_parser: CutOutParser = cut_out_parser.CutOutParser()
+
+        if geojson_path is None:
+            return calculation_state_enum.CalculationState.ERROR_INVALID_CUT_OUT_DATA, "Invalid geojson file"
+
+        try:
+            cells: GeoDataFrame = co_parser.parse_cutout_file(geojson_path)
+        except IllegalCutOutException as err:
+            return calculation_state_enum.CalculationState.ERROR_INVALID_CUT_OUT_DATA, str(err.args[0])
+
+        # Delete result files
+        result_folder: Path = calculation_phase_utility.get_checkpoints_folder_path_from_phase\
+            (configuration_manager, calculation_phase_enum.CalculationPhase.ATTRACTIVITY_PHASE)
+        deleter: FileDeletion = file_deletion.FileDeletion()
+        deleter.reset_folder(result_folder)
+
+        # Iterate over all traffic cells and generate the attractivities
+        index: int
+        row: Series
+        for index, row in cells.iterrows():
```

Comment:
IDK, i did it differently(iterating over all fiels in a folder) i don't know how hard is it to support doing it both ways.
Maybe need to change it, but @GloeStoe knows if he needs to change it, dependent on how he implements process manager.

Reply:
taken the note